integrations/gitlab/gitlab_integration/gitlab_service.py

```diff
@@ -556,3 +560,65 @@ def get_entities_diff(
         logger.info(f"Found {len(entities_after)} entities in the current state")
 
         return entities_before, entities_after
+
+    async def get_all_files_in_project(
+        self, project: Project, path: str
+    ) -> typing.AsyncIterator[List[dict[str, Any]]]:
+        branch = project.default_branch
+        try:
+            file_paths = self._get_file_paths(project, path, branch)
+            logger.debug(
+                f"Found {len(file_paths)} files in project {project.path_with_namespace} files: {file_paths}"
+            )
+            files = []
+            for i, file_path in enumerate(file_paths):
+                try:
+                    project_file = project.files.get(file_path=file_path, ref=branch)
+                    parsed_file = self._process_project_file(project_file)
+                    project_file_dict = project_file.asdict()
+                    if parsed_file:
+                        project_file_dict["content"] = (
+                            parsed_file  # update the content with the parsed content. Useful for JSON and YAML files that can be further processed using itemsToParse
+                        )
+                        files.append(
+                            {"file": project_file_dict, "repo": project.asdict()}
+                        )
+
+                    # Check if the batch size is reached
+                    if (i + 1) % BATCH_SIZE == 0:
+                        yield files
+                        files = []  # Reset the batch
```

Comment:
you already use `files` as a list, so why not checking the length of the files list instead of using `enumerate` ?